torch/distributed/fsdp/_exec_order_utils.py

```diff
@@ -153,19 +151,17 @@ def record_post_forward(self, handles: List[FlatParamHandle]) -> None:
         iteration with the expectation that the recorded order is reset in
         :meth:`next_iter`.
         """
-        if not handles:
+        if not handle:
             return
-        handles_key = tuple(handles)
+        handles_key = handle
         # Only record the first usage of a handles key
         if handles_key in self.handles_to_post_forward_order_index:
             return
         index = len(self.handles_post_forward_order)
         self.handles_to_post_forward_order_index[handles_key] = index
         self.handles_post_forward_order.append(handles_key)
 
-    def record_pre_forward(
-        self, handles: List[FlatParamHandle], is_training: bool
-    ) -> None:
+    def record_pre_forward(self, handle: FlatParamHandle, is_training: bool) -> None:
```

Comment:
```suggestion
    def record_pre_forward(self, handle: Optional[FlatParamHandle], is_training: bool) -> None:
```
since we may actually pass `state._handle` directly, which may be `None`